### ISSUE TITLE
`prefer-describe-function-title` - describe title must match file name case

### ISSUE THREAD

**Issue by ValentinGurkov** [vitest-dev/eslint-plugin-vitest#693]
Hello,

I want to say that I love the idea behind `prefer-describe-function-title`, so I enabled it immediately! 
However, I couldn't get it to work. After some debugging, I found out that the exported describe title must batch the file name case due to:
```ts
const describedTitle = argument.value;
if (!context.filename.includes(`${describedTitle}.`)) {
  return;
}
```
I have 2 cases where this will not work, and I hope you agree with me:

1. For React components, a popular convention is to use `kebab-case` for file names and `PascalCase` for exported components, e.g. `button.tsx` that exports `Button` and the test file equivalent would be `button.test.tsx`.
```tsx
import { Button } from '@/components/button.js';
import { describe } from 'vitest';

describe('Button', () => {
....
```
2. Utility files that export many utility functions - I am not sure what is the best practice here, but if every utility function is tested inside its own `describe` block on top level, then this condition will also not pass.
```tsx
import { a, b } from '@/utils';
import { describe } from 'vitest';

describe('a', () => {
    ...
});

describe('b', () => {
    ...
});
```

I'd love it if you believe this rule should also work for both scenarios

**Comment by veritem** [comment]
@JoshuaKGoldberg, is there any specific reason why the described title must match the file name?

**Comment by JoshuaKGoldberg** [comment]
Ha, I was nervous the rule would be rejected for being too broad. So I put in the constraint there in the hopes it would make the rule seem less overly opinionated. +1 from me on this request 🙂
